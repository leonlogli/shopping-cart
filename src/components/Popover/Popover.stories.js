import React from 'react'

import Popover from '.'

export default {
  title: 'Popover',
}

export const Default = () => (
  <Popover open>
    <div style={{ margin: 16 }}>Pane</div>
  </Popover>
)
