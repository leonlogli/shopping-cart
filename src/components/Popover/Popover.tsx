import React from 'react'
import MuiPopover, {
  PopoverProps as MuiPopoverProps,
  PopoverOrigin,
} from '@material-ui/core/Popover'

export type PopoverProps = Omit<
  MuiPopoverProps,
  'open' | 'anchorOrigin' | 'transformOrigin' | 'onClose'
> & {
  /**
   * If `true`, the popover is visible.
   */
  open?: boolean
  anchorOrigin?: Partial<PopoverOrigin>
  transformOrigin?: Partial<PopoverOrigin>
  onClose?: (event?: React.MouseEvent<HTMLElement>) => void
}

const Popover = (props: PopoverProps) => {
  const {
    anchorOrigin,
    transformOrigin,
    children,
    anchorEl,
    open = !!anchorEl,
    ...other
  } = props

  return (
    <MuiPopover
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
        ...anchorOrigin,
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'center',
        ...transformOrigin,
      }}
      anchorEl={anchorEl}
      open={open}
      {...(other as any)}
    >
      {children}
    </MuiPopover>
  )
}

export default Popover
export { Popover }
