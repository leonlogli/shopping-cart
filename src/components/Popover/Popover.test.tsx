import React from 'react'
import { mount } from 'enzyme'

import Popover from '.'

describe('<Popover />', () => {
  const wrapper = mount(<Popover open>Child</Popover>)

  it('should render children', () => {
    expect(wrapper.text()).toEqual('Child')
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
