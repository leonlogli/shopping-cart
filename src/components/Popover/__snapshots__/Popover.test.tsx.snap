// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should match a snapshot 1`] = `
<Popover
  open={true}
>
  <WithStyles(ForwardRef(Popover))
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    open={true}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <ForwardRef(Popover)
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "bottom",
        }
      }
      classes={
        Object {
          "paper": "MuiPopover-paper",
          "root": "MuiPopover-root",
        }
      }
      open={true}
      transformOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
    >
      <ForwardRef(Modal)
        BackdropProps={
          Object {
            "invisible": true,
          }
        }
        className="MuiPopover-root"
        open={true}
      >
        <ForwardRef(Portal)
          disablePortal={false}
        >
          <Portal
            containerInfo={
              <body
                style="padding-right: 0px; overflow: hidden;"
              >
                <div
                  class="MuiPopover-root"
                  role="presentation"
                  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
                >
                  <div
                    aria-hidden="true"
                    style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: transparent;"
                  />
                  <div
                    data-test="sentinelStart"
                    tabindex="0"
                  />
                  <div
                    class="MuiPaper-root MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
                    style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
                    tabindex="-1"
                  >
                    Child
                  </div>
                  <div
                    data-test="sentinelEnd"
                    tabindex="0"
                  />
                </div>
              </body>
            }
          >
            <div
              className="MuiPopover-root"
              onKeyDown={[Function]}
              role="presentation"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "fixed",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1300,
                }
              }
            >
              <ForwardRef(SimpleBackdrop)
                invisible={true}
                onClick={[Function]}
                open={true}
              >
                <div
                  aria-hidden={true}
                  onClick={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "transparent",
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "left": 0,
                      "position": "fixed",
                      "right": 0,
                      "top": 0,
                      "zIndex": -1,
                    }
                  }
                />
              </ForwardRef(SimpleBackdrop)>
              <Unstable_TrapFocus
                disableAutoFocus={false}
                disableEnforceFocus={false}
                disableRestoreFocus={false}
                getDoc={[Function]}
                isEnabled={[Function]}
                open={true}
              >
                <div
                  data-test="sentinelStart"
                  tabIndex={0}
                />
                <ForwardRef(Grow)
                  appear={true}
                  in={true}
                  onEnter={[Function]}
                  onEntering={[Function]}
                  onExited={[Function]}
                  tabIndex="-1"
                  timeout="auto"
                >
                  <Transition
                    addEndListener={[Function]}
                    appear={true}
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    tabIndex="-1"
                    timeout={null}
                    unmountOnExit={false}
                  >
                    <WithStyles(ForwardRef(Paper))
                      className="MuiPopover-paper"
                      elevation={8}
                      style={
                        Object {
                          "opacity": 1,
                          "transform": "scale(1, 1)",
                          "visibility": undefined,
                        }
                      }
                      tabIndex="-1"
                    >
                      <ForwardRef(Paper)
                        className="MuiPopover-paper"
                        classes={
                          Object {
                            "elevation0": "MuiPaper-elevation0",
                            "elevation1": "MuiPaper-elevation1",
                            "elevation10": "MuiPaper-elevation10",
                            "elevation11": "MuiPaper-elevation11",
                            "elevation12": "MuiPaper-elevation12",
                            "elevation13": "MuiPaper-elevation13",
                            "elevation14": "MuiPaper-elevation14",
                            "elevation15": "MuiPaper-elevation15",
                            "elevation16": "MuiPaper-elevation16",
                            "elevation17": "MuiPaper-elevation17",
                            "elevation18": "MuiPaper-elevation18",
                            "elevation19": "MuiPaper-elevation19",
                            "elevation2": "MuiPaper-elevation2",
                            "elevation20": "MuiPaper-elevation20",
                            "elevation21": "MuiPaper-elevation21",
                            "elevation22": "MuiPaper-elevation22",
                            "elevation23": "MuiPaper-elevation23",
                            "elevation24": "MuiPaper-elevation24",
                            "elevation3": "MuiPaper-elevation3",
                            "elevation4": "MuiPaper-elevation4",
                            "elevation5": "MuiPaper-elevation5",
                            "elevation6": "MuiPaper-elevation6",
                            "elevation7": "MuiPaper-elevation7",
                            "elevation8": "MuiPaper-elevation8",
                            "elevation9": "MuiPaper-elevation9",
                            "outlined": "MuiPaper-outlined",
                            "root": "MuiPaper-root",
                            "rounded": "MuiPaper-rounded",
                          }
                        }
                        elevation={8}
                        style={
                          Object {
                            "opacity": 1,
                            "transform": "scale(1, 1)",
                            "visibility": undefined,
                          }
                        }
                        tabIndex="-1"
                      >
                        <div
                          className="MuiPaper-root MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
                          style={
                            Object {
                              "opacity": 1,
                              "transform": "scale(1, 1)",
                              "visibility": undefined,
                            }
                          }
                          tabIndex="-1"
                        >
                          Child
                        </div>
                      </ForwardRef(Paper)>
                    </WithStyles(ForwardRef(Paper))>
                  </Transition>
                </ForwardRef(Grow)>
                <div
                  data-test="sentinelEnd"
                  tabIndex={0}
                />
              </Unstable_TrapFocus>
            </div>
          </Portal>
        </ForwardRef(Portal)>
      </ForwardRef(Modal)>
    </ForwardRef(Popover)>
  </WithStyles(ForwardRef(Popover))>
</Popover>
`;
